cli/mender-artifact/write.go

```diff
@@ -509,8 +509,8 @@ func getDeviceSnapshot(c *cli.Context) (string, error) {
 	// First echo to stdout such that we know when ssh connection is
 	// established (password prompt is written to /dev/tty directly,
 	// and hence impossible to detect).
-	args = append(args, "sudo", "-S", "/bin/sh", "-c",
-		`'echo "`+sshInitMagic+`"; mender snapshot dump | cat'`)
+	args = append(args, "sudo", "-S", "-s",
+		`echo "`+sshInitMagic+`" && mender snapshot dump | cat`)
```

Comment:
Probably somewhat theoretical, but I would argue that the original is better here. The old version guarantees that we will get the shell we want, whereas `-s` might not work if the user is using some weird shell.